mailet/base/src/test/java/org/apache/mailet/base/test/FakeMailContext.java

```diff
@@ -63,9 +63,12 @@ public static SentMail fromMail(Mail mail ) throws MessagingException {
             .message(mail.getMessage())
             .state(mail.getState())
             .attributes(buildAttributesMap(mail))
+            .attribute(SMTP_AUTH_USER_ATTRIBUTE_NAME, "true")
```

Comment:
This mail do not correspond to a SMTP sent email. Thus you are lying...

I guess the workaround could be to position and "automaticallySent" mail attribute.

Then in the main configuration, we can relay with "RelayAllowed" attribute (that we will introduce).

Then, again in mailetContainer.xml, we can set "RelayAllowed" attribute depending on our will, and allow relaying JMAP, SMTP or automatically sent emails.

I would propose in every mailetcontainer.xml the following changes:

```
            <mailet match="HasMailAttribute=org.apache.james.jmap.send.MailMetaData.messageId"
                    class="SetMailAttribute">
                <RelayAllowed>true</RelayAllowed>
            </mailet>
            <mailet match="SMTPAuthSuccessful" class="SetMailAttribute">
                <RelayAllowed>true</RelayAllowed>
            </mailet>
            <mailet match="HasMailAttribute=AutomaticallySent" class="SetMailAttribute">
                <RelayAllowed>true</RelayAllowed>
            </mailet>
            <mailet match="HasMailAttribute=RelayAllowed" class="RemoteDelivery">
...
            </mailet>
```

This way:
 - We are not lying
 - We are giving more control to the admin